rust/worker/src/execution/operators/prefetch.rs

```diff
@@ -0,0 +1,196 @@
+use thiserror::Error;
+use tonic::async_trait;
+
+use crate::{
+    blockstore::provider::BlockfileProvider,
+    errors::{ChromaError, ErrorCodes},
+    execution::operator::Operator,
+    segment::record_segment::RecordSegmentReader,
+    types::Segment,
+};
+
+#[derive(Debug)]
+pub(crate) struct OffsetIdToDataKeys {
+    pub(crate) keys: Vec<u32>,
+}
+
+#[derive(Debug)]
+pub(crate) struct UserIdToOffsetIdKeys {
+    pub(crate) keys: Vec<String>,
```

Comment:
These keysets could be quite large / are we worried about copy costs?

Reply:
We are worried indeed. Just that we are not taking any action right now since the solution isn't obvious to me immediately.

Reply:
We are worried indeed. Just that we are not taking any action right now. We are not yet using prefetches for user_id_to_offset_id anywhere. I need to audit callsites to see if I could `move` or `Arc<>` instead of full copy from those callsites. Added a todo.

Reply:
haha coolsg